File: php/src/Advertisements/User/UI/Http/SignUpMemberController.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\User\UI\Http;

use Demo\App\Advertisements\User\Application\Command\SignUpMember\SignUpMemberCommand;
use Demo\App\Advertisements\User\Application\Command\SignUpMember\SignUpMemberUseCase;
use Demo\App\Common\Exceptions\BoundedContextException;
use Demo\App\Common\UI\CommonController;
use Demo\App\Framework\FrameworkRequest;
use Demo\App\Framework\FrameworkResponse;
use Demo\App\Framework\SecurityUser\FrameworkSecurityService;

final class SignUpMemberController extends CommonController
{
    public function __construct(
        private SignUpMemberUseCase $useCase,
        private FrameworkSecurityService $securityService,
    ) {}

    public function request(FrameworkRequest $request): FrameworkResponse
    {
        // get role from manager_id request
        $user = $this->securityService->getSecurityUserFromRequest($request);

        if (null == $user || !$user->role() == 'admin') {
            return $this->processUnauthorizedResponse();
        }

        try {
            $command = new SignUpMemberCommand(
                $user->id(),
                $user->role(),
                ($request->content())['id'],
                ($request->content())['email'],
                ($request->content())['password'],
                ($request->content())['memberNumber'],
                ($request->content())['civicCenterId'],
            );

            $this->useCase->execute($command);

            return $this->processSuccessfulCreateCommand();
        } catch (BoundedContextException $exception) {
            return $this->processDomainOrApplicationExceptionResponse($exception);
        } catch (\Throwable $exception) {
            return $this->processGenericException($exception);
        }
    }
}

File: kotlin/src/main/kotlin/advertisements/user/ui/http/SignUpMemberController.kt
package advertisements.user.ui.http

import advertisements.user.application.command.signupmember.SignUpMemberCommand
import advertisements.user.application.command.signupmember.SignUpMemberUseCase
import common.application.ElementNotFoundException
import common.exceptions.BoundedContextException
import common.ui.http.CommonController
import framework.FrameworkRequest
import framework.FrameworkResponse
import framework.securityuser.FrameworkSecurityService

class SignUpMemberController(
    private val useCase: SignUpMemberUseCase,
    private val securityService: FrameworkSecurityService,
): CommonController() {

    override fun execute(request: FrameworkRequest, pathValues: Map<String, String>): FrameworkResponse {
        try {
            val user = securityService.getSecurityUserFromRequest(request)!!

            if (user.role != "admin") {
                return processUnauthorizedResponse()
            }

            useCase.execute(
                SignUpMemberCommand(
                    user.id,
                    user.role,
                    request.content["id"]!!,
                    request.content["email"]!!,
                    request.content["password"]!!,
                    request.content["memberNumber"]!!,
                    request.content["civicCenterId"]!!
                )
            )

            return processSuccessfulCreateCommand()
        } catch (e: ElementNotFoundException) {
            return processNotFoundCommand(e)
        } catch (e: BoundedContextException) {
            return processApplicationOrDomainException(e)
        } catch (e: Exception) {
            return processGenericException(e)
        }
    }
}
